CRM/Core/DAO.php

```diff
@@ -3007,9 +3007,43 @@ public static function getSelectWhereClause($tableAlias = NULL) {
     return $clauses;
   }
 
+  /**
+   * Check whether action can be performed on a given record.
+   *
+   * Dispatches to internal BAO function + hook_civicrm_checkAccess
+   *
+   * @param string $action
+   *   Corresponds to APIv4 action names
+   * @param array $record
+   *   Array of all known values for the record
+   * @param null $contactID
+   *   User id
```

Comment:
I think this should be clarified as the contact is whose permissions are being checked (otherwise on some api it could be the contact who we want to know about permission to)